### ISSUE TITLE
dom0-provided kernels are out of date (again)

### ISSUE THREAD

**Issue by DemiMarie** [QubesOS/qubes-issues#6024]
**Qubes OS version**
`Qubes release 4.0 (R4.0)`

**Affected component(s) or functionality**
dom0-provided kernels

**Brief summary**
`kernel-qubes-vm` is ten versions out of date (4.19.132 vs 4.19.142).  `kernel-latest-qubes-vm` is four versions out of date (5.8.1 vs 5.8.5).

**To Reproduce**
1. Run `rpm -qv kernel-qubes-vm kernel-latest-qubes-vm` in dom0
2. Go to `https://www.kernel.org` and look at what the latest kernel versions from upstream are.

**Expected behavior**
Qubes is at most one or two kernel version out of date

**Actual behavior**
Qubes is several kernel versions out of date

**Screenshots**
N/A

**Additional context**
This does not appear to be due to lack of feedback on testing packages, as I am on the latest testing packages.

**Solutions you've tried**
I can use in-VM kernels, but I first need to make sure that the XSA-155 frontend fixes made it into mainline.

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**
https://www.kernel.org

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**
#4846

**Comment by marmarek** [comment]
The bot fetching new kernels was temporarily offline, but it is back again (see https://github.com/QubesOS/qubes-linux-kernel/pulls)

**Comment by DemiMarie** [comment]
@marmarek worth noting that this problem has happened again.

**Comment by DemiMarie** [comment]
We are now between 3 and 4 versions out of date.

**Comment by DemiMarie** [comment]
I suspect we need a better process.  What if we periodically poll upstream instead of relying on manual approval for each new version?

**Comment by fepitre** [comment]
Currently, it's my bot who takes care of fetching new version, config and building for testing. The only manual step remaining is @marmarek's merge. This rule applies for every repository. We are currently thinking if we could find a way for kernel to have automatic merging but that requires sanity checks. The issue here is @marmarek's schedule and time already massively overloaded...

**Comment by DemiMarie** [comment]
@fepitre what if we required that the only automerged PRs were those that were signed and only bumped the version number?

**Comment by fepitre** [comment]
> @fepitre what if we required that the only automerged PRs were those that were signed and only bumped the version number?

It can work for LTS kernels because the autoupdater does not fetch for newer config. But it's not the case for `kernel-latest`. It tries to get the most updated config from Fedora. I wonder if we cannot find a way to mitigate possible broken/bad config. A possibility could be to test all the kernel PRs into openQA to check that it will not break something. It has the advantage to test and validate before automatic merging. @marmarek what do you think?

**Comment by DemiMarie** [comment]
dom0-provided kernels are now again out of date

**Comment by fepitre** [comment]
If you mean `kernel-latest`: http://lists.infradead.org/pipermail/linux-nvme/2020-November/021204.html

**Comment by DemiMarie** [comment]
That shouldn’t be an issue for `kernel-latest-qubes-vm`, though.

**Comment by fepitre** [comment]
It's too risky to merge such critical behavior. I'm ready to bet that someone would ran into this issue and without any previous dom0 backup kernel...

**Comment by DemiMarie** [comment]
What I meant is that we could build `kernel-latest-qubes-vm` from a different source than `kernel-latest`.

**Comment by DemiMarie** [comment]
`kernel-qubes-vm` is also three versions out of date.

**Comment by marmarek** [comment]
Fresh non-latest kernel was built just yesterday. Don't expect kernel release will land in stable repo 5 minutes after being published.

**Comment by marmarek** [comment]
Hint: you can see recent builds in https://github.com/QubesOS/updates-status/issues

**Comment by DemiMarie** [comment]
@marmarek sorry, I have testing repos enabled, but I forgot that the “updates are available” flag doesn’t fire if the UpdateVM is off.

**Comment by DemiMarie** [comment]
I believe that `kernel` is now two versions out of date compared to upstream.

**Comment by marmarek** [comment]
> I believe that `kernel` is now two versions out of date compared to upstream.

Possibly. Let me repeat: don't expect kernel release will land in stable repo 5 minutes after being published. The process of uploading kernel package to current-testing and only later to current (with at least 7 days delay) is on purpose - for example to prevent issues like https://github.com/QubesOS/qubes-issues/issues/6311 hitting users running just stable ("current") repos enabled. But also, our process allows to move current-testing -> current only the latest package there, so when we upload next version to current-testing, it "resets the timer" - which is fine if the that older version is buggy and shouldn't be moved to current, but otherwise undesirable.

And BTW allowing to move older version is undesirable - the purpose of current-testing repository is to help packages for testing by those who can, not just to introduce artificial delay. Non-latest packages won't receive more testing even if they will sit there longer, because package managers will choose more recent version instead.

In short: yes, kernel package have and will have some weeks delay. It is intentional. You bumping up this issue over and over is counter-productive, we already have notifications about new versions in form of automatic PRs to linux-kernel repository.
